### PR TITLE
Status fix for SNMP

### PR DESCRIPTION
**Pull request by Vzor-**
Fixes SNMP status mismatch with cups

see #1017

### CODE REVIEW

File: src/qz/printer/status/CupsStatusHandler.java [hunk 1]
```diff
@@ -64,6 +65,7 @@ private synchronized void getNotifications() {
                 int attrCount = cups.ippGetCount(jobStateReasonsAttr);
                 for (int i = 0;  i < attrCount; i++) {
                     String reason = cups.ippGetString(jobStateReasonsAttr, i, "");
+                    reason = statusFix(reason);
```
Comment (tresf) [inline src/qz/printer/status/CupsStatusHandler.java:68]: Why not just do this in the map?

Comment (Vzor-) [reply src/qz/printer/status/CupsStatusHandler.java:68]: I wanted to apply the fix as close to the cups side of things a possible. That way the status will already be changed for the duplicate detection.

Comment (tresf) [reply src/qz/printer/status/CupsStatusHandler.java:68]: The map's job is to parse these reasons into enumerated types.  I seems best there, no (and it's only one line lower 😉 )

File: src/qz/printer/status/printer/CupsPrinterStatusMap.java [hunk 2]
```diff
@@ -68,6 +66,9 @@ public enum CupsPrinterStatusMap implements NativeStatus.NativeMap {
     CUPS_INSECURE_FILTER(REASON, NativePrinterStatus.SERVER_UNKNOWN), // "cups-insecure-filter"
     CUPS_MISSING_FILTER(REASON, NativePrinterStatus.ERROR), // "cups-missing-filter"
 
+    // SNMP warning with no existing CUPS definition
```
Comment (tresf) [inline src/qz/printer/status/printer/CupsPrinterStatusMap.java:69]: Minor, but this should be `// SNMP statuses with no existing CUPS definition`

File: src/qz/printer/status/printer/CupsPrinterStatusMap.java [hunk 3]
```diff
@@ -918,4 +921,15 @@ public NativePrinterStatus getParent() {
     public Object getRawCode() {
         return name().toLowerCase(Locale.ENGLISH).replace("_", "-");
     }
+
+    // Bugfix for issue #1017
+    // Removes redundant '-warning' '-report' from SNMP originated statuses, with some exceptions
+    private static String statusFix(String cupsString) {
+        ArrayList exceptions = new ArrayList(Arrays.asList("offline-report", "cups-insecure-filter-warning", "cups-missing-filter-warning"));
```
Comment (tresf) [inline src/qz/printer/status/printer/CupsPrinterStatusMap.java:928]: Please convert exceptions from the enums.

Comment (tresf) [reply src/qz/printer/status/printer/CupsPrinterStatusMap.java:928]: ... arguably, you could programatically create this list too.  I'm going to take a swing at a small refactor and send what I come up with.

**Comment by tresf** [comment]
@Vzor- can you please do some basic testing so that we can get this merged?

File: src/qz/printer/status/printer/CupsPrinterStatusMap.java [hunk 4]
```diff
@@ -918,6 +908,32 @@ public static NativePrinterStatus matchState(String state) {
         return sortedStateLookupTable.getOrDefault(state, NativePrinterStatus.UNMAPPED);
     }
 
+    public static Status createStatus(String reason, String state, String printer) {
+        NativePrinterStatus cupsPrinterStatus = matchReason(reason);
+
```
Comment (tresf) [inline src/qz/printer/status/printer/CupsPrinterStatusMap.java:913]: The comment was lost in this refactor: `// Edge-case for snmp statuses`

**Pull request merged**